packages/desktop-client/src/components/reports/reports/GetCardData.tsx

```diff
@@ -31,19 +33,67 @@ function ErrorFallback() {
   );
 }
 
+function convertFromDate(
+  interval: string | undefined,
+): 'dayFromDate' | 'monthFromDate' | 'yearFromDate' {
+  switch (interval) {
+    case 'Monthly':
+      return 'monthFromDate';
+    case 'Yearly':
+      return 'yearFromDate';
+    default:
+      return 'dayFromDate';
+  }
+}
+
+function convertRangeInclusive(
+  interval: string | undefined,
+): 'dayRangeInclusive' | 'rangeInclusive' | 'yearRangeInclusive' {
+  switch (interval) {
+    case 'Monthly':
+      return 'rangeInclusive';
+    case 'Yearly':
+      return 'yearRangeInclusive';
+    default:
+      return 'dayRangeInclusive';
+  }
+}
+
 export function GetCardData({
   report,
   payees,
   accounts,
   categories,
   earliestTransaction,
+  firstDayOfWeekIdx,
 }: {
   report: CustomReportEntity;
   payees: PayeeEntity[];
   accounts: AccountEntity[];
   categories: { list: CategoryEntity[]; grouped: CategoryGroupEntity[] };
   earliestTransaction: string;
+  firstDayOfWeekIdx?: LocalPrefs['firstDayOfWeekIdx'];
 }) {
+  const fromDate = convertFromDate(report.interval);
+  const rangeInclusive = convertRangeInclusive(report.interval);
+
+  let dateStart;
+  let dateEnd;
```

Comment:
dateStart and dateEnd are redefined as constants when the report doesn't have a static date a little below. Likely using a different variable name here will help reduce confusion about them.

Also, do we need this section to be below the call to getLiveRange and use the outcomes of that?

Reply:
Yes and yes. Done and done.